diff --git a/core/Makefile b/core/Makefile
index dc88fa9..1b055d4 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -17,7 +17,7 @@ OBJECTS+=quassel-net.o quassel-msgs.o quassel-cmds.o
 OBJECTS+=irssi/network-openssl.o
 OBJECTS+=quassel-fe-window.o quassel-fe-level.o quassel-cfg.o
 
-CFLAGS=-std=gnu11 -Wall -Wextra -Werror -g -O2 $(IRSSI_CFLAGS) -Wmissing-prototypes -Wmissing-declarations
+CFLAGS+=$(shell pkg-config --cflags quasselc) -std=gnu11 -Wall -Wextra -Werror $(IRSSI_CFLAGS) -Wmissing-prototypes -Wmissing-declarations
 
 TARGET=libquassel_core.so
 
@@ -31,7 +31,7 @@ irssi/network-openssl.o: CFLAGS:=$(IRSSI_CFLAGS)
 quasselc-connector.o: CFLAGS:=$(CFLAGS) -Ilib
 
-$(TARGET): $(OBJECTS) $(LIBQUASSEL)
+$(TARGET): $(OBJECTS)
-	gcc -shared $^ -o $@ -lz
+	gcc -shared $^ -o $@ -lz $(shell pkg-config --libs quasselc) $(LDFLAGS)
 
 install: $(TARGET)
 	$(INSTALL) -d $(IRSSI_LIB)/modules
diff --git a/core/quassel-irssi.h b/core/quassel-irssi.h
index 9554915..153fedb 100644
--- a/core/quassel-irssi.h
+++ b/core/quassel-irssi.h
@@ -1,7 +1,7 @@
 #ifndef QUASSEL_IRSSI_H
 #define QUASSEL_IRSSI_H
 
-#include "lib/export.h"
+#include "export.h"
 
 #define STRUCT_SERVER_CONNECT_REC struct Quassel_SERVER_CONNECT_REC_s 
 typedef struct Quassel_SERVER_CONNECT_REC_s {
